lib/vmpooler/api.rb

```diff
@@ -33,20 +10,54 @@ def initialize
         require File.expand_path(File.join(File.dirname(__FILE__), 'api', lib))
       end
     end
+    # Load dashboard components
+    begin
+      require 'dashboard'
+    rescue LoadError
+      require File.expand_path(File.join(File.dirname(__FILE__), 'dashboard'))
+    end
 
-    use Vmpooler::API::Dashboard
-    use Vmpooler::API::Reroute
-    use Vmpooler::API::V1
-
-    def configure(config, redis, metrics)
+    def self.execute(torun, config, redis, metrics)
       self.settings.set :config, config
-      self.settings.set :redis, redis
+      self.settings.set :redis, redis unless redis.nil?
       self.settings.set :metrics, metrics
       self.settings.set :checkoutlock, Mutex.new
-    end
 
-    def execute!
-      self.settings.run!
+      # Deflating in all situations
+      # https://www.schneems.com/2017/11/08/80-smaller-rails-footprint-with-rack-deflate/
+      use Rack::Deflater
+
+      # not_found clause placed here to fix rspec test issue.
+      not_found do
+        content_type :json
+
+        result = {
+          ok: false
+        }
+
+        JSON.pretty_generate(result)
+      end
+
+      if metrics.respond_to?(:setup_prometheus_metrics)
+        # Prometheus metrics are only setup if actually specified
+        # in the config file.
+        metrics.setup_prometheus_metrics
+
+        # Using customised collector that filters out hostnames on API paths
+        require File.expand_path(File.join(File.dirname(__FILE__), 'metrics', 'collector_vmpooler'))
+        use Prometheus::Middleware::CollectorVmpooler, metrics_prefix: "#{metrics.metrics_prefix}_http"
```

Comment:
I would suggest renaming this to more accurately represent where the code lives and such. Maybe make the use line something like `use Vmpooler::Promstats::Middleware::Collector` and update the file, module, and class names accordingly

Reply:
Done